src/data/list/pairwise.lean

```diff
@@ -0,0 +1,311 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+import data.list.basic
+
+open nat function
+
+universes u v
+
+variables {α : Type u} {β : Type v}
+
+namespace list
+
+/- pairwise relation (generalized no duplicate) -/
+
+section pairwise
```

Comment:
```suggestion
```

Could you maybe move the definition of the `pairwise` relation here as well?

Reply:
Moved.